kie-assets-plugin/src/main/java/org/kie/mojos/GenerateProjectMojo.java

```diff
@@ -324,4 +350,62 @@ private Path getPathToPom(ProjectDefinition definition, ProjectStructure structu
+     */
+    private void addMavenConfigFile(ProjectDefinition definition, ProjectStructure structure) throws IOException {
+        Formatter content = new Formatter();
+        determineUserSettings(structure).ifPresent(file -> content.format("-s %s%n", file.getAbsolutePath()));
+        determineLocalRepo(structure).ifPresent(file -> content.format("-Dmaven.repo.local=%s%n", file.getAbsolutePath()));
+        Path projectDir = GeneratedProjectUtils.getOutputDirectoryForGeneratedProject(outputDirectory.toPath(), definition, structure);
+        Path mvnDir = projectDir.resolve(".mvn");
+        Files.createDirectories(mvnDir);
+        Path mavenConfigFile = mvnDir.resolve("maven.config");
+        Files.writeString(mavenConfigFile, content.toString(), StandardCharsets.UTF_8, StandardOpenOption.CREATE);
+    }
+
+    /**
+     * Determines user settings to use for project generation. By default, it will use the same user settings the main build (the one running
+     * kie-assets-library plugin) is using. If a custom {@linkplain ProjectGeneration#settingsFile} is provided
+     * in the {@linkplain ProjectStructure}, it will override the settings from the main build. localRepository fragment is ignored as this
+     * is controlled by {@linkplain ProjectGeneration#useSeparateRepository()}.
+     *
+     * @param structure project structure to determine user settings
+     * @return user settings file to use for the project generation
+     */
+    private Optional<File> determineUserSettings(ProjectStructure structure) {
+        File settingsXml = null;
+        if (mavenSession != null && mavenSession.getRequest() != null) { // for tests
+            settingsXml = mavenSession.getRequest().getUserSettingsFile();
+        }
+        File customSettingsXml = structure.getGenerate().getSettingsFile();
+        if (customSettingsXml != null) {
+            settingsXml = customSettingsXml;
+        }
+        return Optional.ofNullable(settingsXml);
+    }
+
+    /**
+     * Determines local repository to use for project generation. By default, it will use the same local repository the main build (the one running
+     * kie-assets-library plugin) is using. If a {@linkplain ProjectGeneration#useSeparateRepository()} is set to 'true'
+     * in the {@linkplain ProjectStructure}, it will override the local repository from the main build.
+     *
+     * @param structure project structure to determine local repository
+     * @return local repository to use for the project generation
+     */
+    private Optional<File> determineLocalRepo(ProjectStructure structure) {
+        File localRepo = null;
+        if (mavenSession != null && mavenSession.getRequest() != null) { // for tests
+            localRepo = mavenSession.getRequest().getLocalRepositoryPath();
+        }
+        if (structure.getGenerate().useSeparateRepository()) {
+            localRepo = new File(outputDirectory, structure.getId() + "-local-repo");
```

Comment:
Would the file name combination be a good candidate for a utils method? e.g. in GeneratedProjectUtils, my concern is to rely on custom string concatenation if the dir-name is needed also in other placed (so far just tests, but who knows).

Reply:
Sure, it's a good idea.